Source/WebCore/rendering/RenderLayer.cpp

```diff
@@ -1271,9 +1271,12 @@ void RenderLayer::recursiveUpdateLayerPositionsAfterScroll(RenderGeometryMap* ge
     } else if (!renderer().view().frameView().platformWidget()) {
         // When ScrollView's m_paintsEntireContents flag flips due to layer backing changes, the repaint area transitions from
         // visual to layout overflow. When this happens the cached repaint rects become invalid and they need to be recomputed (see webkit.org/b/188121).
-        // Check that our cached rects are correct.
+        // Check that our cached rects are correct for Mac WK2 and WPE ports.
+        // Skip these assertions for Mac WK1, GTK and Windows ports which don't enable ForceCompositingMode.
+#if !PLATFORM(GTK) && !PLATFORM(WIN)
         ASSERT_IMPLIES(m_repaintRectsValid, m_repaintRects.clippedOverflowRect == renderer().clippedOverflowRectForRepaint(renderer().containerForRepaint().renderer.get()));
         ASSERT_IMPLIES(m_repaintRectsValid, m_repaintRects.outlineBoundsRect == renderer().outlineBoundsForRepaint(renderer().containerForRepaint().renderer.get()));
+#endif
```

Comment:
```suggestion
        if (renderer().settings().forceCompositingMode()) {
            ASSERT_IMPLIES(m_repaintRectsValid, m_repaintRects.clippedOverflowRect == renderer().clippedOverflowRectForRepaint(renderer().containerForRepaint().renderer.get()));
            ASSERT_IMPLIES(m_repaintRectsValid, m_repaintRects.outlineBoundsRect == renderer().outlineBoundsForRepaint(renderer().containerForRepaint().renderer.get()));
        }
```
This probably makes more sense

Reply:
Good idea. Will fix. Thank you.

Reply:
Fixed